test/unit-test/test-Model.js

```diff
@@ -6836,6 +6836,596 @@ describe('Unit Test/Model Test', function() {
+      let model = await nn.createModel(options);
+      let output = {type: nn.TENSOR_FLOAT32, dimensions: [1, 2, 2, 1]};
+
+      model.addOperand(output);
+
+      model.addOperation(nn.BATCH_TO_SPACE_ND, [], [0]);
+      model.identifyInputsAndOutputs([], [0]);
+      await model.finish();
+
+      let compilation = await model.createCompilation();
+      compilation.setPreference(getPreferenceCode(options.prefer));
+      await compilation.finish();
+
+      let execution = await compilation.createExecution();
+      execution.setOutput(0, new Float32Array(product(output.dimensions)));
+      await assertThrowsAsync(async() => {
+        await execution.startCompute();
+      });
+    });
+
+    it('raise error when setting 0 output for "BATCH_TO_SPACE_ND" operation', async function() {
+      let model = await nn.createModel(options);
+      let input0 = {type: nn.TENSOR_FLOAT32, dimensions: [4, 1, 1, 1]};
+      let input1 = {type: nn.TENSOR_INT32, dimensions: [2]};
+
+      model.addOperand(input0);
+      model.addOperand(input1);
+
+      model.setOperandValue(1, new Int32Array([2, 2]));
+      model.addOperation(nn.BATCH_TO_SPACE_ND, [0, 1], []);
+      model.identifyInputsAndOutputs([0], []);
+      await model.finish();
+
+      let compilation = await model.createCompilation();
+      compilation.setPreference(getPreferenceCode(options.prefer));
+      await compilation.finish();
+
+      let execution = await compilation.createExecution();
+      execution.setInput(0, new Float32Array([1, 2, 3, 4]));
+      await assertThrowsAsync(async() => {
+        await execution.startCompute();
+      });
+    });
+
+    it('raise error when the block value < 1 for "BATCH_TO_SPACE_ND" operation', async function() {
+      let model = await nn.createModel(options);
+      let input0 = {type: nn.TENSOR_FLOAT32, dimensions: [4, 1, 1, 1]};
+      let input1 = {type: nn.TENSOR_FLOAT32, dimensions: [2]};
+      let output = {type: nn.TENSOR_FLOAT32, dimensions: [1, 2, 2, 1]};
+
+      model.addOperand(input0);
+      model.addOperand(input1);
+      model.addOperand(output);
+
+      model.setOperandValue(1, new Float32Array([0.2, 0.2]));
+      model.addOperation(nn.BATCH_TO_SPACE_ND, [0, 1], [2]);
+      model.identifyInputsAndOutputs([0], [2]);
+      await model.finish();
+
+      let compilation = await model.createCompilation();
+      compilation.setPreference(getPreferenceCode(options.prefer));
+      await compilation.finish();
+
+      let execution = await compilation.createExecution();
+      execution.setInput(0, new Float32Array([1, 2, 3, 4]));
+      execution.setOutput(0, new Float32Array(product(output.dimensions)));
+      await assertThrowsAsync(async() => {
+        await execution.startCompute();
+      });
+    });
+
+    it('"4-D tensor as input0 that is TENSOR_FLOAT32 type and 1-D tensor as input1 that is TENSOR_INT32 type" are ok for "TRANSPOSE" operation', function() {
```

Comment:
According to https://github.com/intel/webml-polyfill/blob/master/src/nn/Enums.js#L1224,
```
1: An optional 1-D Tensor of {@link TENSOR_INT32}, the permutation of  the dimensions of the input tensor. 
```
input1 is an optional tensor, 
so please add some cases with input having only input0 for TRANSPOSE op, thanks.